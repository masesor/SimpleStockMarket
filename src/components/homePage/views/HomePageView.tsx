import * as React from 'react';

import './HomePageView.css';
import { TradeTable } from 'src/components/tradeTable';
import { Grid, Button, WithStyles, Theme, createStyles } from '@material-ui/core';
import { isSmartphone } from 'src/responsive';
import { WithWidth } from '@material-ui/core/withWidth';
import { NewTradeFormDialog } from 'src/components/newTradeFormDialog';

export const styles = (theme:Theme) => createStyles({
    root: {
        padding: theme.spacing.unit * 10,
    },

    mobileRoot: {
        paddingTop: 50,
        paddingLeft: 15,
        paddingRight: 15,
    },

    button: {
        marginBottom: 15,
    },
});

export interface IProps extends WithStyles<typeof styles>, WithWidth {
}

export interface IDispatchProps {
    toggleNewTradeFormDialog:() => any;
}

export const HomePageView = (props:IProps & IDispatchProps) => (
    <Grid
        container
        className={isSmartphone(props.width) ? props.classes.mobileRoot : props.classes.root}
        alignItems={'flex-start'}
        justify={'flex-start'}
    >
        <NewTradeFormDialog />
        <Grid item xs={12}>
            <Button
                className={props.classes.button}
                variant="contained"
                color="secondary"
                onClick={() => props.toggleNewTradeFormDialog()}>
                Book Trade
        </Button>
        </Grid>
        <Grid item xs={12}>
            <TradeTable />
        </Grid>
    </Grid>
);
