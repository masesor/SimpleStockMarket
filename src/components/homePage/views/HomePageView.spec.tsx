import * as React from 'react';

import { HomePageView } from './HomePageView';
import { shallow } from 'enzyme';

describe('<HomePageView />', () => {
  it('renders the component as expected', () => {
    const classes:any = {
      root: {
        padding: 10,
      },

      mobileRoot: {
        paddingTop: 50,
        paddingLeft: 15,
        paddingRight: 15,
      },

      button: {
        marginBottom: 15,
      },
    };
    const wrapper = shallow(
      <HomePageView
        toggleNewTradeFormDialog={jest.fn()}
        classes={classes}
        width={undefined}
      />);
    expect(wrapper).toMatchSnapshot();
  });
});
