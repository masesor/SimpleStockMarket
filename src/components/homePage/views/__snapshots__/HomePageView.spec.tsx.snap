// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    classes={
        Object {
            "button": Object {
              "marginBottom": 15,
            },
            "mobileRoot": Object {
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 50,
            },
            "root": Object {
              "padding": 10,
            },
          }
    }
    toggleNewTradeFormDialog={[Function]}
    width={undefined}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "alignItems": "flex-start",
      "children": Array [
        <WithRoot />,
        <WithStyles(Grid)
          item={true}
          xs={12}
>
          <WithStyles(Button)
                    className={
                              Object {
                                        "marginBottom": 15,
                                      }
                    }
                    color="secondary"
                    onClick={[Function]}
                    variant="contained"
          >
                    Book Trade
          </WithStyles(Button)>
</WithStyles(Grid)>,
        <WithStyles(Grid)
          item={true}
          xs={12}
>
          <WithRoot />
</WithStyles(Grid)>,
        <WithStyles(Grid)
          item={true}
          xs={12}
>
          <WithRoot />
</WithStyles(Grid)>,
      ],
      "className": Object {
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 50,
      },
      "container": true,
      "justify": "flex-start",
      "spacing": 40,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Button)
            className={
                        Object {
                                    "marginBottom": 15,
                                  }
            }
            color="secondary"
            onClick={[Function]}
            variant="contained"
>
            Book Trade
</WithStyles(Button)>,
          "item": true,
          "xs": 12,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Book Trade",
            "className": Object {
              "marginBottom": 15,
            },
            "color": "secondary",
            "onClick": [Function],
            "variant": "contained",
          },
          "ref": null,
          "rendered": "Book Trade",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithRoot />,
          "item": true,
          "xs": 12,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithRoot />,
          "item": true,
          "xs": 12,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "alignItems": "flex-start",
        "children": Array [
          <WithRoot />,
          <WithStyles(Grid)
            item={true}
            xs={12}
>
            <WithStyles(Button)
                        className={
                                    Object {
                                                "marginBottom": 15,
                                              }
                        }
                        color="secondary"
                        onClick={[Function]}
                        variant="contained"
            >
                        Book Trade
            </WithStyles(Button)>
</WithStyles(Grid)>,
          <WithStyles(Grid)
            item={true}
            xs={12}
>
            <WithRoot />
</WithStyles(Grid)>,
          <WithStyles(Grid)
            item={true}
            xs={12}
>
            <WithRoot />
</WithStyles(Grid)>,
        ],
        "className": Object {
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 50,
        },
        "container": true,
        "justify": "flex-start",
        "spacing": 40,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Button)
              className={
                            Object {
                                          "marginBottom": 15,
                                        }
              }
              color="secondary"
              onClick={[Function]}
              variant="contained"
>
              Book Trade
</WithStyles(Button)>,
            "item": true,
            "xs": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Book Trade",
              "className": Object {
                "marginBottom": 15,
              },
              "color": "secondary",
              "onClick": [Function],
              "variant": "contained",
            },
            "ref": null,
            "rendered": "Book Trade",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithRoot />,
            "item": true,
            "xs": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithRoot />,
            "item": true,
            "xs": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<HomePageView /> renders the component as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    classes={
        Object {
            "button": Object {
              "marginBottom": 15,
            },
            "mobileRoot": Object {
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 50,
            },
            "root": Object {
              "padding": 10,
            },
          }
    }
    toggleNewTradeFormDialog={[Function]}
    width={undefined}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "alignItems": "flex-start",
      "children": Array [
        <WithRoot />,
        <WithStyles(Grid)
          item={true}
          xs={12}
>
          <WithStyles(Button)
                    className={
                              Object {
                                        "marginBottom": 15,
                                      }
                    }
                    color="secondary"
                    onClick={[Function]}
                    variant="contained"
          >
                    Book Trade
          </WithStyles(Button)>
</WithStyles(Grid)>,
        <WithStyles(Grid)
          item={true}
          xs={12}
>
          <WithRoot />
</WithStyles(Grid)>,
        <WithStyles(Grid)
          item={true}
          xs={12}
>
          <WithRoot />
</WithStyles(Grid)>,
      ],
      "className": Object {
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 50,
      },
      "container": true,
      "justify": "flex-start",
      "spacing": 40,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Button)
            className={
                        Object {
                                    "marginBottom": 15,
                                  }
            }
            color="secondary"
            onClick={[Function]}
            variant="contained"
>
            Book Trade
</WithStyles(Button)>,
          "item": true,
          "xs": 12,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Book Trade",
            "className": Object {
              "marginBottom": 15,
            },
            "color": "secondary",
            "onClick": [Function],
            "variant": "contained",
          },
          "ref": null,
          "rendered": "Book Trade",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithRoot />,
          "item": true,
          "xs": 12,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithRoot />,
          "item": true,
          "xs": 12,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "alignItems": "flex-start",
        "children": Array [
          <WithRoot />,
          <WithStyles(Grid)
            item={true}
            xs={12}
>
            <WithStyles(Button)
                        className={
                                    Object {
                                                "marginBottom": 15,
                                              }
                        }
                        color="secondary"
                        onClick={[Function]}
                        variant="contained"
            >
                        Book Trade
            </WithStyles(Button)>
</WithStyles(Grid)>,
          <WithStyles(Grid)
            item={true}
            xs={12}
>
            <WithRoot />
</WithStyles(Grid)>,
          <WithStyles(Grid)
            item={true}
            xs={12}
>
            <WithRoot />
</WithStyles(Grid)>,
        ],
        "className": Object {
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 50,
        },
        "container": true,
        "justify": "flex-start",
        "spacing": 40,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Button)
              className={
                            Object {
                                          "marginBottom": 15,
                                        }
              }
              color="secondary"
              onClick={[Function]}
              variant="contained"
>
              Book Trade
</WithStyles(Button)>,
            "item": true,
            "xs": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Book Trade",
              "className": Object {
                "marginBottom": 15,
              },
              "color": "secondary",
              "onClick": [Function],
              "variant": "contained",
            },
            "ref": null,
            "rendered": "Book Trade",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithRoot />,
            "item": true,
            "xs": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithRoot />,
            "item": true,
            "xs": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
